src/type_checker/checker.rs

```diff
@@ -507,7 +507,7 @@ impl<B: Backend> TypeChecker<B> {
                         .compute_type(&observed.1, typed_fn_env)?
                         .expect("expected a value (TODO: better error)");
 
-                    if !observed_typ.typ.match_expected(&defined.1) {
+                    if !observed_typ.typ.match_expected(&defined.1, true) {
```

Comment:
Similar to the comment to the assignment case above, this is because it allows assigning either constant or not.